src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private int size;
+    private Node<T> head;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> previous;
```

Comment:
There is a condition in readme file: "Your implementation should be based on the array or singly linked list (don't use existing collections)". And your Linked list is double linked.

Reply:
Thank you, did not notice this information